lib/Net/Async/Redis/Cluster.pm

```diff
@@ -218,9 +218,17 @@ sub node_by_host_port {
     return $node;
 }
 
-sub register_moved_slot {
+async sub register_moved_slot {
     my ($self, $slot, $host_port) = @_;
-    my $node = $self->node_by_host_port(split /:/, $host_port);
+    my ($host, $port) = split /:/, $host_port;
+    my $node = $self->node_by_host_port($host, $port);
+     if(!$node) {
+        $log->tracef("Failed to find node %s:%s in the original node list", $host, $port);
+        # Has a replica become a primary?
+        await $self->bootstrap(host => $host, port => $port);
```

Comment:
This might be better handled by calling `apply_slots_from_instance` on one of our existing instances, if any are still valid? It'd save the overhead of the extra connect/scan/drop sequence.

We'd want to refresh our node list and slot allocation anyway after a `MOVED` notification.